docs/resources.adoc

```diff
@@ -126,12 +124,20 @@ You can create as many global shared resources as you want, as long as they have
 [NOTE]
 ====
 We do not support creating shared resources with arguments.
-
 This is because if a test refers to a shared resource with the name "Foo" without arguments, then later another test refers to it with one argument, there is no reasonable way to fulfill that request.
-
 Furthermore, even if this was supported, the behavior would change if the first and second tests ever ran in opposite order, which is very likely when tests are configured to run in parallel.
 ====
 
+== Cleaning Up Resources
+
+Resources will be cleaned up, meaning `close` will be called on them and they will be made eligible for garbage collection, when they are no longer needed:
+
+* for a `@New` parameter, this happens immediately after the test
+* for a `@Shared` parameter with scope `SOURCE_FILE`, this happens when all tests in that source file were executed
+* for a `@Shared` parameter with scope `GLOBAL`, this happens when the entire test suite was executed
+
+Resources will be cached in-memory until then.
+
```

Comment:
I like this, especially how to clarifies what "cleaning up" a resource means, given that the documentation uses both the words "clean" and "close". :)